### PR TITLE
jnp.where: add optional size argument

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #3614; uses #6501

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1704,14 +1704,19 @@ def _where(condition, x=None, y=None):
 At present, JAX does not support JIT-compilation of the single-argument form
 of :py:func:`jax.numpy.where` because its output shape is data-dependent. The
 three-argument form does not have a data-dependent shape and can be JIT-compiled
-successfully.
+successfully. Alternatively, you can specify the optional ``size`` keyword.
+If specified, the first ``size`` True elements will be returned; if there
+are fewer True elements than ``size`` indicates, the index arrays will be
+padded with zeros.
```
Comment (8bitmp3) [inline jax/_src/numpy/lax_numpy.py:1710]: Small suggestion - link the last sentences maybe? They may look more cohesive if together? 🤷‍♂️ 

```suggestion
successfully. Alternatively, you can specify the optional ``size`` keyword:
if specified, the first ``size`` True elements will be returned; and if there
are fewer True elements than ``size`` indicates, the index arrays will be
padded with zeros.
```

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Done, thanks

File: tests/lax_numpy_test.py [hunk 2]
```diff
@@ -4189,6 +4189,10 @@ def testWhereOneArgument(self, shape, dtype):
     args_maker = lambda: [rng(shape, dtype)]
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False)
 
+    # JIT compilation requires specifying the size statically:
+    jnp_fun = lambda x: jnp.where(x, size=np.size(x) // 2)
```
Comment (froystig) [inline tests/lax_numpy_test.py:4193]: Is `size(x) // 2` guaranteed sufficient under all random draws of the argument? If I understand correctly, [that's only the expected size](https://github.com/google/jax/blob/648b5d3265c3f1e28be39195a194981239ef2c6d/jax/test_util.py#L728), and there could be fewer zeros.

Comment (jakevdp) [reply tests/lax_numpy_test.py:4193]: In general `size` may be larger or smaller than the true number, and `jnp.where` should be able to handle that.

**Review by froystig — APPROVED** [review]

**Pull request merged**